buildpack/buildcontext.go

```diff
@@ -0,0 +1,57 @@
+// Package buildpack provides the core infrastructure
+package buildpack
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/lifecycle/log"
+)
+
+const (
+	// SharedContextDir defines a shared context for build- and run-image extensions
+	SharedContextDir = "context"
+)
+
+// ContextInfo captures info about the context used for build- and run-image extensions
+type ContextInfo struct {
+	ExtensionID string
+	Kind        string
+	Path        string
+}
+
+func findContexts(d ExtDescriptor, extOutputDir string, logger log.Logger) ([]ContextInfo, error) {
+	var contexts []ContextInfo
+	var sharedIsProvided bool
+
+	sharedContextDir := filepath.Join(extOutputDir, SharedContextDir)
+	if s, err := os.Stat(sharedContextDir); err == nil && s.IsDir() {
+		sharedIsProvided = true
+		contexts = append(contexts, ContextInfo{
+			ExtensionID: d.Extension.ID,
+			Path:        sharedContextDir,
+		})
+	} else if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	for _, dir := range []string{DockerfileKindRun, DockerfileKindBuild} {
+		contextDir := filepath.Join(extOutputDir, dir)
```

Comment:
Will this not look for the context dir in `<output>/run` or `<output>/build`? Shouldn't it be `<output>/context.run` or `<output>/context.build`?